src/bindings/python/src/pyopenvino/graph/model.cpp

```diff
@@ -259,6 +269,65 @@ void regclass_graph_Model(py::module m) {
             :param name: String to set as model's friendly name.
             :type name: str
             )");
+
+    model.def(py::init([](const ov::OutputVector& results,
+                          const ov::OutputVector& nodes,
+                          const ov::ParameterVector& parameters,
+                          const ov::op::util::VariableVector& variables,
+                          const std::string& name) {
+                  set_tensor_names(parameters);
+                  const auto sinks = cast_to_sink_vector(nodes);
+                  auto model = std::make_shared<ov::Model>(results, sinks, parameters, name);
+                  set_correct_variables_for_assign_ops(model, sinks);
+                  return model;
+              }),
+              py::arg("results"),
+              py::arg("sinks"),
+              py::arg("parameters"),
+              py::arg("variables") = ov::op::util::VariableVector(),
+              py::arg("name") = "",
+              R"(
+            Create user-defined Model which is a representation of a model
+
+            :param results: List of outputs.
+            :type results: List[openvino.runtime.Output]
+            :param sinks: List of Output sink node handles.
+            :type sinks: List[openvino.runtime.Output]
+            :param variables: List of variables.
+            :type variables: List[op.util.Variable]
+            :param name: String to set as model's friendly name.
+            :type name: str
+            )");
+
+    model.def(py::init([](const ov::ResultVector& results,
+                          const ov::OutputVector& nodes,
+                          const ov::ParameterVector& parameters,
+                          const ov::op::util::VariableVector& variables,
```

Comment:
we still need to pass variables to constructor.
If you want to keep this version of constructor where variables can be omitted, may be it is worth to add a check if variablesVector is empty or smthg like this?